### ISSUE TITLE
add diagnostic for unused table aliases

### ISSUE THREAD

**Issue by lewis6991** [EmmyLuaLs/emmylua-analyzer-rust#724]
In Neovim code (and Lua code in general), it's pretty common for aliases to be set up in the file.

```lua
local gsub = string.gsub

local a = gsub(...)
```

However over time, new patches can forget to use these aliases and end up using the full qualified name:

```lua
local gsub = string.gsub

local a = gsub(...)

...

local b = string.gsub(...)
```

Can a diagnostic be implemented to provide a hint that the user should prefer to use the existing alias?

**Comment by CppCXY** [comment]
Such checks need to clearly define their scope. Usages like `text:gsub("xxx")` should not trigger warnings, and variables that are mutable within their scope should also not be warned.

**Comment by CppCXY** [comment]
complete: 6cfce3455cb1c2e04070884696c374c2b900e552

**Comment by lewis6991** [comment]
Need to check for reassignments:

<img width="504" height="148" alt="Image" src="https://github.com/user-attachments/assets/217a50d6-5ed2-4d92-b48a-a4137faa8d72" />

Maybe just restrict this to aliases defined in module scope?

**Comment by CppCXY** [comment]
done f1582662a601a2b930e331d73d20c9e9818fd4a2
